pkg/scheduling/requirements.go

```diff
@@ -219,5 +219,7 @@ func (r Requirements) Labels() map[string]string {
 
 func (r Requirements) String() string {
 	requirements := lo.Reject(r.Values(), func(requirement *Requirement, _ int) bool { return v1alpha5.RestrictedLabels.Has(requirement.Key) })
-	return strings.Join(lo.Map(requirements, func(requirement *Requirement, _ int) string { return requirement.String() }), ", ")
+	stringRequirements := lo.Map(requirements, func(requirement *Requirement, _ int) string { return requirement.String() })
```

Comment:
We don't need this anymore, right?

Reply:
We do need it. The events can still be made to look as new events, even if the dudupe is respected. 
```
  Warning  FailedScheduling  2m19s  karpenter          Failed to schedule pod, incompatible with provisioner "default", daemonset overhead={"cpu":"125m","pods":"2"}, no instance type satisfied resources {"cpu":"125m","nvidia.com/gpu":"1","pods":"3"} and requirements karpenter.sh/capacity-type In [on-demand], karpenter.k8s.aws/instance-category In [c m r], kubernetes.io/os In [linux], karpenter.sh/provisioner-name In [default], kubernetes.io/arch In [amd64] (no instance type which had enough resources and the required offering met the scheduling requirements)
  Warning  FailedScheduling  88s    karpenter          Failed to schedule pod, incompatible with provisioner "default", daemonset overhead={"cpu":"125m","pods":"2"}, no instance type satisfied resources {"cpu":"125m","nvidia.com/gpu":"1","pods":"3"} and requirements kubernetes.io/os In [linux], karpenter.sh/provisioner-name In [default], kubernetes.io/arch In [amd64], karpenter.sh/capacity-type In [on-demand], karpenter.k8s.aws/instance-category In [c m r] (no instance type which had enough resources and the required offering met the scheduling requirements)
```

Reply:
Ah, you mean when there's a describe done on the pod? Makes sense.